pkg/docker/containers-view.jsx

```diff
@@ -185,14 +209,53 @@ var ContainerList = React.createClass({
         this.setState({ containers: containers });
     },
 
+    setNewProblem: function (c_id, url, message) {
+        /* New problem is always displayed, no matter if the same problem is
+         * already shown. It is because user may be interested into dynamic
+         * watching of the problems occurring. After refreshing the site, only
+         * the latest occurrence is displayed.
+         */
+        var known_problems = this.state.problems;
+        if (c_id in known_problems)
+            known_problems[c_id].push([url, message]);
+        else
+            known_problems[c_id] = [[url, message]];
+        this.setState({ problems: known_problems });
+    },
+
+    newProblemOccurred: function (event, problem_path) {
+        var self = this;
+        util.find_container_log(this.problems_client, problem_path, self.setNewProblem);
+    },
+
     componentDidMount: function () {
+        var self = this;
+        this.problems_client = cockpit.dbus('org.freedesktop.problems', { superuser: "try" });
+        this.service = this.problems_client.proxy('org.freedesktop.Problems2', '/org/freedesktop/Problems2');
+        this.problems = this.problems_client.proxies('org.freedesktop.Problems2.Entry', '/org/freedesktop/Problems2/Entry');
+        this.problems.wait(function() {
+            if (typeof self.service.GetSession !== "undefined"){
+                self.service.GetSession().done(function(session_path) {
+                    var session = self.problems_client.proxy('org.freedesktop.Problems2.Session', session_path);
```

Comment:
This proxy isn't necessary either as you just do a single function call on it (this is even easier with `cockpit.dbus.call()`). However, there is now just a fixed static set of proxies, not a dynamically growing and unbounded one any more, so it's good enough now.